# User Input
The world is growing just because of people's input. What if things become static, it will ruin the whole cycle of nature. This example best fit into this topic.
There are two types of softwares in the world static and dynamic. In static there is nothing to input from the user side for example calendar app, and on the other side dynamic softwares are those softwares in which the next option only work when user input somehting for example you have to input the +  operation in order to get the addition of two inputs. Here we are going to use input function for the user interactions.

In [1]:
input()

''

*For getting the input from the user we must display the message about what kind of input we need. Either we need string type of input or integer type of input.*

In [2]:
input("what's your name ")

''

*You must save the input into variable for a good programming practice. You can use this variable at multiple place where you need to ask same question from the user.*

In [4]:
name = input("What's your name? ")
print(name)

umair


**Take two numbers from the user and add it**

In [8]:
num1 = input("first number: ")
num2 = input("second number: ")
addition = num1 + num2
print(addition)

2222


In [9]:
print(type(addition))

<class 'str'>


*I gave the two inputs 22 and 22 but rather than showing 44 it showed me 2222. What is this. This is just because of data type. Data type of these variables are string, thats why it was not added. Lets make it correct. By the way this concept is called type conversion. You can change the type of any variable*  

In [10]:
num1 = int(input("first number: "))
num2 = int(input("second number: "))
addition = num1 + num2
print(addition)

44


*Bravao*

___

# Doubts
- What is type conversion
    - converting one data type into other is called type conversion.

___